credential-provider/src/services/core.service.ts

```diff
@@ -36,6 +42,108 @@ export class CoreService {
     return res;
   }
 
+  /**
+   * Performs the following to create callback URL for interaction-hook containing a new response token
+   *
+   * 1. Verify session_token
+   * 2. Create responseToken
+   * 3. Construct & return callbackUrl for redirect on interaction-hook
+   * @param CreateCallbackUrlArgs
+   * @returns string
+   * @example
+   * await createCallbackUrl({
+   *   session_token: "ONE_TIME_JWT_TOKEN",
+   * })
+   */
+  public async createCallbackUrl(args: CreateCallbackUrlArgs): Promise<string> {
+    const { session_token } = args;
+
+    const secret = await this.getInteractionHookSecret();
+
+    const verifiedJwt = await this.verifyInteractionHookToken({
+      session_token,
+      secret,
+    });
+
+    const responseToken = await this.createInteractionHookResponseToken({
+      session_token,
+      verifiedJwt,
+      secret,
+    });
+
+    /** The verifiedJwt will have redirectUrl inside its payload
+     * which is additional to the types returned by default JWTVerifyResult */
+    const { redirectUrl } = verifiedJwt.payload;
+
+    const callbackUrl = `${redirectUrl}?session_token=${responseToken}`;
+    this.logger.log(
+      `Finished processing Interaction Hook request, user will be redirected to ${callbackUrl}`,
+    );
+    return callbackUrl;
+  }
+
+  public async getInteractionHookSecret(): Promise<Buffer> {
+    const getOpenIdConfigRes = await this.mattrService.getOpenIdConfig({
+      token: this.config.get('MATTR_AUTH_TOKEN'),
+    });
+    const encodedSecret = getOpenIdConfigRes.data.interactionHook.secret;
+    return Buffer.from(encodedSecret, 'base64');
+  }
+
+  public async verifyInteractionHookToken(
+    args: VerifyInteractionHookTokenArgs,
+  ): Promise<JWTVerifyResult> {
+    const { session_token, secret } = args;
+
+    const decoded = decodeJwt(session_token);
+    const audience = decoded.aud as string;
+    const issuer = decoded.iss;
```

Comment:
I noticed I was decoding `session_token` in both `verifyInteractionHookToken` and `createInteractionHookResponseToken` to get `audience` and `issuer` - I've thought about passing them into both functions as args, but then I realised getting these values from a decoded `session_token` is much more reliable, even if that means rewriting 2-3 lines